### ISSUE TITLE
`love.getVersion()` doesn't return revision information anymore

### ISSUE THREAD

**Issue by tpimh** [love2d/love#2118]
This topic for some reason provoked a heated discussion in Discord yesterday, so I would like to first state that this is a very minor concern, I don't have a problem with how things work now, and I am documenting my findings here only for your information, this is in no way a bait for unhealthy discussion.

Right now LÖVE is using GitHub Releases for the distribution of binaries. One problem with GitHub Releases is that they do not preserve history: hotfixes overwrite older builds, which become lost. For example, the link https://github.com/love2d/love/releases/download/11.5/love-11.5-win64.zip used to point at 3 different files at different times, different developers might have slightly different builds depending on when they downloaded the release.

It's extremely hard to tell from runtime which hotfix is currently running. It would probably involve finding out the path to the running executable, then checking it's meta information. This would be platform dependent and extremely hacky.

Revision number returned by `love.getVersion()` is always set to 0 since the release of LÖVE 11. This number can be reused to indicate the hotfix number or build date (e.g. "last two digits of the year" .. "week number with leading zero"). This is just a suggestion, not call to action.

In conclusion, I would like to reiterate: I am fine with how things work now, just wishing that in the future versions of LÖVE telling hotfix versions apart was easier.

**Comment by slime73** [comment]
Here's what I said in discord:

> overall, having hotfixes like this was a fairly unique situation for 11.5. If it stops being a unique situation we'll formalize it in a way that fits our specific needs, but I don't expect it to continue
